src/utils/Hardware.ts

```diff
@@ -0,0 +1,246 @@
+import fetchJSON from './fetchJSON'
+
+interface AcessibleControllerAPI {
+  connected: boolean
+}
+interface BatteryAPI {
+  charging: boolean
+  level: number // Number between 0–1
+}
+interface CardReaderAPI {
+  connected: boolean
+}
+interface PrinterAPI {
+  connected: boolean
+}
+
+/**
+ * Defines the API for accessing hardware status.
+ */
+export interface Hardware {
+  /**
+   * Reads Accessible Controller status
+   */
+  readAccesssibleControllerStatus(): Promise<AcessibleControllerAPI>
+
+  /**
+   * Sets Accessible Controller connected
+   */
+  setAccesssibleControllerConnected(connected: boolean): Promise<void>
+
+  /**
+   * Reads Battery status
+   */
+  readBatteryStatus(): Promise<BatteryAPI>
+
+  /**
+   * Sets Battery charging
+   */
+  setBatteryCharging(charging: boolean): Promise<void>
+
+  /**
+   * Sets Battery level.  Number between 0–1.
+   */
+  setBatteryLevel(level: number): Promise<void>
```

Comment:
These `set*` methods don’t represent operations we can actually perform in non-test scenarios, i.e. the battery level is what it is. I’d say these should not be present on the `Hardware` interface but should be defined on `MemoryHardware`. Unless I’m missing something?